File: C#/Memento/Exercise/CanvasState.cs
namespace Memento.Exercise;

public sealed class CanvasState
{
    public string Content { get; set; }
    public string Color { get; set; }
    public string Border { get; set; }

    public CanvasState(string content, string color, string border)
    {
        Content = content;
        Color = color;
        Border = border;
    }
}

File: typescript/memento/examples/CanvasState.ts
export class CanvasState {
  constructor(
    private content: string,
    private color: string,
    private border: string
  ) {
    this.content = content;
    this.color = color;
    this.border = border;
  }

  getContent(): string {
    return this.content;
  }

  setContent(content: string) {
    this.content = content;
  }

  getColor(): string {
    return this.color;
  }

  setColor(color: string) {
    this.color = color;
  }

  getBorder(): string {
    return this.border;
  }

  setBorder(border: string) {
    this.border = border;
  }
}
